### PR TITLE
added parquet loader

### PR DESCRIPTION
**Pull request by chris-pickford**
1. Added PArquet laoder
2. added interface
3. added tests

### CODE REVIEW

File: datadayessentials/data_retrieval/_base.py [hunk 1]
```diff
@@ -147,6 +147,14 @@ def __init__(self, authentication: IAuthentication):
     def load(self) -> Any:
         pass
 
+class IParquetLoader(IDataFrameLoader):
+    def __init__(self, authentication: IAuthentication):
+        pass
+
+    @abstractmethod
+    def load(self) -> Any:
```
Comment (whitew1994WW) [inline datadayessentials/data_retrieval/_base.py:155]: Guessing it should return an object of DataFrame type as it inherits from dataframeloader?

File: datadayessentials/data_retrieval/_load_data.py [hunk 2]
```diff
@@ -532,3 +533,56 @@ def load(self, blob: BlobLocation) -> dict:
...
+        cacher = DataCacher(str(blob) + ".parquet", properties.last_modified)
+        if cacher.is_file_in_cache() and self.use_cache:
+            return cacher.get_df_from_cache()
+        else:
+            download = file_client.download_file()
+            buffer = BytesIO(download.read())
+            buffer.seek(0)
+            df = pd.read_parquet(buffer)
+            cacher.save_df_to_cache(df)
+            return df
```
Comment (Colin-Hanley) [inline datadayessentials/data_retrieval/_load_data.py:588]: I can see this not working as intended. But lets wait to see how the final save solution looks like.

**Review by Colin-Hanley — APPROVED** [review]
approved. Comments in test are acceptable since its an area to improve on.

**Review by whitew1994WW — CHANGES_REQUESTED** [review]
Not sure about the exact use case for the loader, but we might want to test the loading of specific partitions of parquet

File: datadayessentials/data_retrieval/_base.py (hunk 1 above)
Comment (chris-pickford) [reply datadayessentials/data_retrieval/_base.py:155]: Good catch

**Pull request merged**